assets/javascript/logic.js

```diff
@@ -157,3 +157,37 @@ function scrubKeys(object) {
   }
   return object;
 }
+
+function saveRecipeToCurrentTab(searchResultIndex) {
+  var newRecipe;
+  newRecipe = searchResults[searchResultIndex];
+  var label = recipeTabs[parseInt($("#card-tab-select").attr("value"))];
+  saveRecipe(newRecipe, label);
+  // copied from above
+  $("#content").empty();
+  $("#recipe-display-modal")
+    .detach()
+    .appendTo($("#storage"));
+  var index = recipeTabs.indexOf(newRecipe.tab);
+  $("#tab-label").text(recipeTabs[index]);
+  $("#tab-select").attr("value", index + "");
+  updateRecipeBox();
```

Comment:
updateRecipeBox() now takes two arguments -- a recipe object and an array of keys for that object. 

I think you can just delete the call to updateRecipeBox() because saveRecipe(), which you call earlier, will take care of refreshing the recipes in a tab.

Reply:
done